controlplane/pkg/tests/nsmd_test_utils.go

```diff
@@ -31,69 +31,94 @@ import (
 	"google.golang.org/grpc"
 )
 
+const (
+	Master = "master"
+	Worker = "worker"
+)
+
+type sharedStorage struct {
+	services  map[string]*registry.NetworkService
+	managers  map[string]*registry.NetworkServiceManager
+	endpoints map[string]*registry.NetworkServiceEndpoint
+}
+
+func newSharedStorage() *sharedStorage {
+	return &sharedStorage{
+		services:  make(map[string]*registry.NetworkService),
+		endpoints: make(map[string]*registry.NetworkServiceEndpoint),
+		managers:  make(map[string]*registry.NetworkServiceManager),
+	}
+}
+
 type nsmdTestServiceDiscovery struct {
 	apiRegistry *testApiRegistry
-
-	services   map[string]*registry.NetworkService
-	managers   map[string]*registry.NetworkServiceManager
-	endpoints  map[string]*registry.NetworkServiceEndpoint
-	nsmCounter int
+	storage     *sharedStorage
```

Comment:
Nit: is this really a storage?

Reply:
This is class that stores all custom resources for all service registries, so I think storage is appropriate name